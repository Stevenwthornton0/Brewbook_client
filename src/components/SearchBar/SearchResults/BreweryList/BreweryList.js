import React, { Component } from 'react';
import BreweryItem from './BreweryItem/BreweryItem';
import BreweryListContext from '../../../../contexts/BreweryListContext';

class BreweryList extends Component {
    static contextType = BreweryListContext;

    renderBreweries() {
        let { breweryList } = this.context
        breweryList = breweryList.filter(brewery => brewery.brewery_type !== 'planning')
            return breweryList.map(brewery => 
                <BreweryItem 
                    key={brewery.id}
                    brewery={brewery}
                />
            )
    }

    render() {
        return (
            <div>{this.renderBreweries()}</div>
        )
    }
}

export default BreweryList;