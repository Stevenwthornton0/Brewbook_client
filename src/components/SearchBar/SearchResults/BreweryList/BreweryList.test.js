import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import BreweryList from './BreweryList';

Enzyme.configure({ adapter: new Adapter() })

describe(`BreweryList smoke test`, () => {
    it(`renders without crashing`, () => {
        const wrapper = mount(<BreweryList />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})