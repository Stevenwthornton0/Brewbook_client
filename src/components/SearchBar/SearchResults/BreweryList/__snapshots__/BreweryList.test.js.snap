// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryList smoke test renders without crashing 1`] = `
<BreweryList>
  <div />
</BreweryList>
`;
